plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -193,6 +233,28 @@ func validateOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta
 	return nil
 }
 
+func validateScheduling(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
+	if runtimeClass == nil || runtimeClass.Scheduling == nil {
+		return nil
+	}
+
+	nodeScheduling := &node.Scheduling{}
+	if err := nodev1beta1.Convert_v1beta1_Scheduling_To_node_Scheduling(runtimeClass.Scheduling, nodeScheduling, nil); err != nil {
+		return err
+	}
+
+	// If the Scheduling contains a conflict node selector which has the same key but differnet value, reject the pod
+	runtimeNodeSelector := nodeScheduling.NodeSelector
+	newNodeSelector := pod.Spec.NodeSelector
+	for key, value := range runtimeNodeSelector {
+		if podValue, ok := newNodeSelector[key]; ok && podValue != value {
```

Comment:
we should drop this func, as this code is mostly duplicated on `setScheduling()`